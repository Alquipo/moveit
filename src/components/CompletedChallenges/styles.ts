import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;

    margin: 3.5rem 0;
    padding-bottom: 1rem;
    border-bottom: 1px solid #d7d8da;

    font-weight: 500;

    color: ${theme.colors.text};

    span:first-child {
      font-size: 1.25rem;
    }

    span:last-child {
      font-size: 1.5rem;
    }
  `}
`
