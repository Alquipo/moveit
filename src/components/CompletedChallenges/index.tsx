import { useContext } from 'react'
import { ChallengesContext } from 'contexts/ChallengesContext'
import * as S from './styles'

const CompletedChallenges = () => {
  const { challengesCompleted } = useContext(ChallengesContext)

  return (
    <S.Wrapper>
      <span>Desafios completos</span>
      <span>{challengesCompleted}</span>
    </S.Wrapper>
  )
}

export default CompletedChallenges
