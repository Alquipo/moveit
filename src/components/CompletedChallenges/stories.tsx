import { Story, Meta } from '@storybook/react/types-6-0'
import CompletedChallenges from '.'

export default {
  title: 'CompletedChallenges',
  component: CompletedChallenges
} as Meta

export const Default: Story = () => <CompletedChallenges />
