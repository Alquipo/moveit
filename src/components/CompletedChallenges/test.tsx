import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import CompletedChallenges from '.'

describe('<CompletedChallenges />', () => {
  it('should render the CompletedChallenges correctly', () => {
    const { container } = renderWithTheme(<CompletedChallenges />)

    expect(screen.getByText(/Desafios completos/i)).toBeInTheDocument()

    expect(screen.getByText(/5/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
