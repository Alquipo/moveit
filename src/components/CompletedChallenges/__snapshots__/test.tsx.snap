// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompletedChallenges /> should render the CompletedChallenges correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 3.5rem 0;
  padding-bottom: 1rem;
  border-bottom: 1px solid #d7d8da;
  font-weight: 500;
  color: #EEEEEE;
  -webkit-transition: color 0.3s ease-in-out;
  transition: color 0.3s ease-in-out;
}

.c0 span:first-child {
  font-size: 1.25rem;
}

.c0 span:last-child {
  font-size: 1.5rem;
}

<div
  class="c0"
>
  <span>
    Desafios completos
  </span>
  <span>
    5
  </span>
</div>
`;
